### PR TITLE
Param ui

### PR DESCRIPTION
**Pull request by k1o0**
- [x] Added new integrated parameter editor
- [x] Added test class for editor
- [x] Added test class for Parameters
- [x] Added performance test
- [x] Comment the ParamEditor, FieldPanel and ConditionTable classes
- [ ] Deleted ParamEditor_old
- [x] Restored param description and unit functionality
- [x] Restored label formatting

### CODE REVIEW

**Comment by k1o0** [comment]
- Building new ParamEditor window with parameters loaded takes < 0.4s and to load in a new set of parameters < 0.3s
- There are numerous bug fixes uncovered by the test class
- numRepeats can now be a global parameter.  This changes the way the trial conditions are repeated giving the user more flexibility
- Trial conditions can now be fixed, i.e. the user can choose not to randomize the trials within the GUI
- There are now tests for the ParamEditor class and Parameters class
- ParamEditor and Parameters classes are more thoroughly documented

Still left to do for another day:
- [ ] Properly integrate string params: currently they're converted to char when made conditional.  This isn't necessary.
- [ ] Add test for set values method
- [ ] Add method for sorting trial conditions by a column
- [ ] Add method for setting specific trial condition order (i.e. adding conditionIdx param functionality into SignalsExp)

File: +eui/ConditionPanel.m [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+  methods
+    function obj = ConditionPanel(f, ParamEditor, varargin)
+      % FIELDPANEL Panel UI for Conditional parameters 
+      %  Input f may be a figure or other UI container object
+      %  ParamEditor is a handle to an eui.ParamEditor object.
+      % 
+      % See also EUI.PARAMEDITOR, EUI.FIELDPANEL
+      obj.ParamEditor = ParamEditor;
+      obj.UIPanel = uix.VBox('Parent', f);
+%       obj.UIPanel.BackgroundColor = 'white';
```
Comment (jaib1) [inline +eui/ConditionPanel.m]: can this be removed?

Comment (k1o0) [reply +eui/ConditionPanel.m]: You may recall that the editor was originally white, which I think looks nicer.  MC is still all grey by default.  I can't be arsed to change everything to white so the short answer is yes, let's remove all of these BackgroundColor lines.

File: +eui/FieldPanel.m [hunk 2]
```diff
@@ -0,0 +1,229 @@
+classdef FieldPanel < handle
```
Comment (jaib1) [inline +eui/FieldPanel.m]: why call this 'FieldPanel' instead of something like 'GlobalPanel'?

File: +eui/MControl.m [hunk 3]
```diff
@@ -247,10 +247,11 @@ function saveParamProfile(obj) % Called by 'Save...' button press, save a new pa
     end
     
     function loadParamProfile(obj, profile)
+      tic
```
Comment (jaib1) [inline +eui/MControl.m]: can the 'tic' and subsequent 'toc' be removed?

File: +eui/FieldPanel.m (hunk 2 above)
Comment (k1o0) [reply +eui/FieldPanel.m]: This is what Chris originally called it.  It's a panel that generates UI fields.  I suppose it's more functionally clear than 'global' which is a less concrete name

File: +eui/MControl.m (hunk 3 above)
Comment (k1o0) [reply +eui/MControl.m]: Shit, yes.  I think there's a toc at the bottom of the script.  This was done before I wrote the performance test class

File: tests/ParamEditorPerfTest.m [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    function setup(testCase)
+      % Hide figures and add teardown function to restore settings
+      testCase.FigureVisibleDefault = get(0,'DefaultFigureVisible');
+      set(0,'DefaultFigureVisible','off');
+      testCase.addTeardown(@set, 0, 'DefaultFigureVisible', testCase.FigureVisibleDefault);
+            
+      % Loads validation data
+      %  Graph data is a cell array where each element is the graph number
+      %  (1:3) and within each element is a cell of X- and Y- axis values
+      %  respecively
```
Comment (jaib1) [inline tests/ParamEditorPerfTest.m]: typo

File: +eui/ConditionPanel.m (hunk 1 above)
Comment (jaib1) [reply +eui/ConditionPanel.m]: ha hmm well you're right the white background does look nicer...

**Comment by k1o0** [comment]
1. The ConditionPanel and FieldPanel classes aren't really standalone so standalone tests aren't useful.  All the functionality of these two classes is tested by the ParamEditor test.
2. Okay
3. These errors aren't really informative, could we run then test together tomorrow?

**Comment by k1o0** [comment]
Before merging, let's make sure the tests run on a real rig, double-check everything works properly within mc and also delete the +eui/ParamEditor_old.m file.

**Comment by k1o0** [comment]
Oh and the .github/config.yml didn't seem to do anything.  I suppose we just delete it, unless I made a typo somewhere?

**Comment by jaib1** [comment]
Ok, I may have time to test it on the behaviour rigs after upgrading the hardware, while I'm installing the new MATLABs. Would late afternoon work for you? Yeah the .yml file looks right... I'll also look into it

File: +eui/ConditionPanel.m [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+    MakeGlobalButton
+    % UIControl button for setting multiple table cells at once
+    SetValuesButton
+    % Indicies of selected table cells  as array [row, column;...] of each
+    % selected cell
+    SelectedCells 
+  end
+  
+  methods
+    function obj = ConditionPanel(f, ParamEditor, varargin)
```
Comment (jaib1) [inline +eui/ConditionPanel.m]: why is there a varargin arg here?

**Comment by jaib1** [comment]
I've reorganized the test folder structure and think I found a way to safely test with "dat.paths" (and any other folder/file we may need to shadow"). Let me know what you think

**Comment by k1o0** [comment]
[Example of shared fixtures](https://uk.mathworks.com/help/matlab/ref/matlab.unittest.fixtures.pathfixture-class.html#bt0cnij-2).  All it really affords you is fewer lines of code as you don't have to manually add the taredown routine.

**Comment by jaib1** [comment]
ok, i will merge this now, but keep parameditor_old around, and keep this branch open

**Pull request merged**